### ISSUE TITLE
Positioning issues with long pages, translate values differ on page by page basis

### ISSUE THREAD

**Issue by rhysmatthew** [dixonandmoe/rellax#251]
I'm finding that parallaxed elements further down pages, depending on page lengths can get wildly different translate values (to the point where they're not even visible sometimes). Even with center: true

I find this library works great for items near the top of pages but I'm finding it really tricky when working with elements that could be near the bottom of an article for example or in the footer (my screenshots example).

I love this library and feel it has a great use case, but I would love to see an option whereby you can set a scroll container... where it only translates the element within the bounds of that scroll container. I.e. in my case here it'd be the footer element, I only want to start translating my elements when the footer is in view, not based on body scroll which can be vastly different on each page. It seems quite expensive in terms of performance too as there is no need to translate elements out of view all of the time.

I know this could take a great deal of work to look into or might not be possible, which is understandable. Just thought I'd put my thoughts down to see if anyone else experiences the same or has any solutions. I've used this library on 4-5 projects now and had the same issue every time. 

![longer-page](https://github.com/dixonandmoe/rellax/assets/77306946/237145a1-91b4-436e-8719-09348d83b361)
![shorter-page](https://github.com/dixonandmoe/rellax/assets/77306946/85184466-f9d3-47e2-915a-c9c175685df2)

**Comment by ADTC** [comment]
This exact problem makes Rellax unusable for me. In your case, you're still applying it to decorative elements, where such issues are more forgivable, and you can work around the problem a little (although I'm totally with you that it's extremely annoying to have to do so).

In my case, I'm applying it to elements that are content-based, and it's absolutely unacceptable how Rellax is translating them completely outside the container of the content element into a totally random position.

The [demo website](https://dixonandmoe.com/rellax/) is highly misleading. If you notice, as you scroll down, you'll find some diamond shapes floating over the docs.

https://github.com/user-attachments/assets/58527239-faf1-46d9-896f-3732b16a7b47

Disable JavaScript and you'll reveal the hidden truth: They're strategically placed in positions far below what you'd expect (viewport in / viewport out):

<img width="1302" alt="Image" src="https://github.com/user-attachments/assets/94530aec-f85b-4f83-9d0a-66e5819f2a78" />

**Comment by rhysmatthew** [comment]
> This exact problem makes Rellax unusable for me. In your case, you're still applying it to decorative elements, where such issues are more forgivable, and you can work around the problem a little (although I'm totally with you that it's extremely annoying to have to do so).
> 
> In my case, I'm applying it to elements that are content-based, and it's absolutely unacceptable how Rellax is translating them completely outside the container of the content element into a totally random position.
> 
> The [demo website](https://dixonandmoe.com/rellax/) is highly misleading. If you notice, as you scroll down, you'll find some diamond shapes floating over the docs.
> 
>  Screen.Recording.2025-03-10.at.10.25.09.PM.mp4 
> Disable JavaScript and you'll reveal the hidden truth: They're strategically placed in positions far below what you'd expect (viewport in / viewport out):
> 
> <img alt="Image" width="1302" src="https://private-user-images.githubusercontent.com/6047296/420952414-94530aec-f85b-4f83-9d0a-66e5819f2a78.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE4NjY0NzMsIm5iZiI6MTc0MTg2NjE3MywicGF0aCI6Ii82MDQ3Mjk2LzQyMDk1MjQxNC05NDUzMGFlYy1mODViLTRmODMtOWQwYS02NmU1ODE5ZjJhNzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTNUMTE0MjUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2RlNTBkYTcxMDVkOWU0MzZjMDQ5OTMzNzkyOTE5M2MwMGZjODNhYTZkNDI5YTMyN2UwZjJlN2Y0MjJmYTc4MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.MKxv3EoEFqZm5oyTt6AoMaQ3llmqd31wwBy22YpjDmQ">

Exactly, glad I'm not the only one struggling with this. It's a great plugin except for this quirk, I feel like it needs rethinking from the ground up / the ability to set parent or relative containers so that elements can be anywhere on a page.

**Comment by rhysmatthew** [comment]
I know that Custom Wrappers (`wrapper` option) exist but I did try using this and didn't get the desired effect, i.e. no movement at all. I'm guessing this is intended for a different purpose like a `<main>` element that scrolls instead of the body etc.
